tests/rest/client/v2_alpha/test_register.py

```diff
@@ -863,7 +863,7 @@ def make_homeserver(self, reactor, clock):
         config["account_validity"] = {"enabled": False}
 
         self.hs = self.setup_test_homeserver(config=config)
-        self.hs.config.account_validity_period = self.validity_period
+        self.hs.get_datastore()._account_validity_period = self.validity_period
```

Comment:
Would a better way to do this be to define `default_config`?

Reply:
`hs.config.account_validity_period` still won't be set even if `account_validity.period` if defined in the homeserver config though, right?

Reply:
We talked a bit in #synapse-dev about this. The reason this test was working before is because we were reading `period` and `startup_job_max_delta` from the homeserver config file during test execution.

Now we're reading account validity config values during homeserver initialisation, which will happen during the `self.setup_test_homeserver` call. *Before* setting those homeserver config options.

I've added a comment to explain this, but this test probably still needs a separate look. (For instance, why does the docstring of `test_background_job` say it does the same thing as `test_background_job`??).